modules/autocmd.nix

```diff
@@ -42,7 +42,7 @@ with lib;
     let
       inherit (config) autoGroups autoCmd;
     in
-    mkIf (autoGroups != { } || autoCmd != { }) {
+    mkIf (autoGroups != { } || autoCmd != [ ]) {
```

Comment:
Unrelated fix

Reply:
All this fixes was to remove empty extra empty lines from output. This fixes a condition that is guarding `extraConfigLuaPost` from producing empty string.

Reply:
Got it.

It is unrelated to the "use mkIf instead of optionalString" commit, but not unrelated to this PR if it solves the bigger issue.